tensorboard/webapp/plugins/npmi/util/umap.ts

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```

Comment:
Test is missing in this module.

Reply:
I added some tests, but I am not sure if this is the way to go with asynchronous functions. Maybe you can give me some feedback on that.